cors.go

```diff
@@ -0,0 +1,90 @@
+package go_dfms_gateway
+
+import (
+	"strings"
+
+	"github.com/valyala/fasthttp"
+)
+
+func NewCors(allowedMethods []string, allowedHeaders []string, allowedOrigins []string) *cors {
+	return &cors{
+		Enable:         true,
+		AllowedMethods: allowedMethods,
+		AllowedHeaders: allowedHeaders,
+		AllowedOrigins: allowedOrigins,
+	}
+}
+
+func DefaultCors() *cors {
+	return &cors{
+		Enable:         false,
+		AllowedMethods: []string{"*"},
+		AllowedHeaders: []string{"*"},
+		AllowedOrigins: []string{"*"},
+	}
+}
+
+type cors struct {
+	Enable         bool
+	AllowedMethods []string
+	AllowedHeaders []string
+	AllowedOrigins []string
+}
+
+func (c *cors) check(ctx *fasthttp.RequestCtx) error {
+	if !c.Enable {
+		return nil
+	}
+
+	// check origin
+	origin := string(ctx.Request.Header.Peek("Origin"))
+	if origin != "" && isAllowed(c.AllowedOrigins, origin) {
+		ctx.Response.Header.Set("Access-Control-Allow-Origin", origin)
+	}
+
+	//ctx.Response.Header.Set("Access-Control-Allow-Methods", "Content-type")
+	//ctx.Response.Header.Set("Access-Control-Allow-Headers", "GET")
+
+	//check method
+	requestMethod := string(ctx.Request.Header.Peek("Access-Control-Request-Method"))
+	if requestMethod != "" || isAllowed(c.AllowedMethods, requestMethod) {
+		ctx.Response.Header.Set("Access-Control-Allow-Methods", strings.Join(c.AllowedMethods, ","))
+	}
```

Comment:
What is the purpose of checking methods two times: here and in the func that `newMiddleware` returns?